fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -1810,7 +1811,20 @@ private List<CreateReplicaTask> buildCreateReplicaTasks(long dbId, OlapTable tab
             if (table.isCloudNativeTableOrMaterializedView()) {
                 long primaryComputeNodeId = -1;
                 try {
-                    primaryComputeNodeId = ((LakeTablet) tablet).getPrimaryComputeNodeId();
+                    if (Config.only_use_compute_node) {
+                        // it is ok for cloudnative table to build tablet in any warehouse
+                        Warehouse warehouse = GlobalStateMgr.getCurrentWarehouseMgr().
+                                getWarehouse(WarehouseManager.DEFAULT_WAREHOUSE_NAME);
```

Comment:
how about the warehousemgr provide `getWarehouse()` interface and returns the default warehouse. The caller doesn't care about what the default ware house name is.